### ISSUE TITLE
[css-display-3] parent box vs anon boxes

### ISSUE THREAD

**Issue by prjnt** [w3c/csswg-drafts#3259]
https://drafts.csswg.org/css-display-3/#css-parent-box says:
<blockquote>The <dfn>parent box</dfn> of an element’s principal box is (except in the case of run-in) the principal box of its nearest ancestor element that generates a box.</blockquote>
<p>The above implies that an anonymous box is not the parent box of any element's principal box, whereas I believe that it is desired that anonymous boxes can have an element's principal box as a child box, and that the anonymous box should sometimes furthermore establish the containing block for an element's principal box.

**Comment by prjnt** [comment]
Renaming this concept to something other than "parent box" might well be an appropriate fix, assuming that there is indeed prose that intends to refer to this concept rather than parent node in box tree and that such prose can be updated to use the new name.

[Regarding anonymous boxes and percentages, note that the quoted CSS2 visuren.html text is limited to anonymous *block* boxes: other anonymous boxes, such as anonymous table cells, still establish the containing block in the sense of being used as the basis for resolving percentage widths.]

**Comment by fantasai** [comment]
No, it's just an error. The definition should match @prjnt's expectations in the OP. :)

**Comment by fantasai** [comment]
OK @prjnt , I posted some edits. Lmk if it looks correct now or if you have any suggestions for improvement!

**Comment by fantasai** [comment]
Based on feedback from @Loirooriol in issue 3158, the text now reads

> In constructing the box tree, boxes generated by an element are descendants of the principal box of any ancestor elements. In the general case, the direct parent box of an element’s principal box is the principal box of its nearest ancestor element that generates a box; however, there are some exceptions, such as for run-in boxes, display types that generate multiple container boxes (such as tables), and intervening anonymous boxes.
